### DISCUSSION TITLE
Data in Configservice

### DISCUSSION THREAD

**Discussion by Vishwa1705** [aws/aws-sdk-go-v2#1881] [Q&A]
Hi there, 

Service : Configsevrice
Function : BatchGetResourceConfig

I've noticed that for some items such as EC2, ARN is empty in the response.

Also the "AwsRegion" field in response, SDK says - "The Region where resource resides", but we are getting wrong region.

Ex : EC2 Instance is in us-east-2, but response says it's in us-east-1.

Can you please help me to get the correct region where the resources reside in?

Thanks.

**Comment by RanVaknin** [comment]
Hi @c0sm1k ,

This is hard to know without seeing the way you set up your request but if I had to guess you are not specifying the correct region in your config:
```go
	cfg, err := config.LoadDefaultConfig(context.TODO(), config.WithRegion("<YOUR DESIRED REGION GOES HERE>"))
```

Here is my working code:

```go
func main() {
	cfg, err := config.LoadDefaultConfig(context.TODO(), config.WithRegion("<YOUR DESIRED REGION GOES HERE>"))
	if err != nil {
		log.Fatalf("unable to load SDK config, %v", err)
	}

	res, err := client.BatchGetResourceConfig(context.Background(), &configservice.BatchGetResourceConfigInput{
		ResourceKeys: []types.ResourceKey{
			{
				ResourceId:   aws.String(myInstanceID),
				ResourceType: "AWS::EC2::Instance",
			},
		},
	})
	if err != nil {
		panic(err)
	}

	for _, key := range res.BaseConfigurationItems {
		fmt.Println(*key.AwsRegion)
	}
}
```

Let me know if this helps.
Thanks,
Ran~

**Comment by Vishwa1705** [comment]
Hi @RanVaknin 

Thanks for responding quick.

We have set configuration, created client through NewFromConfig(<aws.Config>) (This is done).
We are using AWS Config for only one purpose - to get all resources, along with their Id and ARN. (This is also done).
Now we need to group them their region basis. So we are referring to the field (AwsRegion) that comes with response.

We need the region where resource is located. How can we get it?

```
for _, eachResource := range res.BaseConfigurationItems{
  fmt.Println(eachResource.Arn) <-- Resource ARN, sometimes returns empty (for EC2 items)
  fmt.Println(eachResource.ResourceId) <-- Resource ID 
  fmt.Println(eachResource.AwsRegion) <-- This is supposed to give Resource's Region, Instead giving the one passed in cfg.
```

Thanks.

**Reply by RanVaknin** [reply]
Hi @c0sm1k ,

Can you please share your complete code example?
